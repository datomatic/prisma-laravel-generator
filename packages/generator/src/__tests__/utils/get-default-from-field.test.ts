import {getMongoSample, getSample} from '../__fixtures__/get-sample';
import getDefaultFromField from '../../utils/get-default-from-field';

test('getDefaultFromField', async () => {
  const {dmmf} = await getSample();

  for (const modelInfo of dmmf.datamodel.models) {
    for (const fieldInfo of modelInfo.fields) {
      expect(
        getDefaultFromField(fieldInfo, dmmf.datamodel.enums),
      ).toMatchSnapshot(`${modelInfo.name} - ${fieldInfo.name}`);
    }
  }
});

test('getDefaultFromField - mongodb', async () => {
  const {dmmf} = await getMongoSample();

  for (const modelInfo of dmmf.datamodel.models) {
    for (const fieldInfo of modelInfo.fields) {
      expect(
        getDefaultFromField(fieldInfo, dmmf.datamodel.enums, 'mongodb'),
      ).toMatchSnapshot(`${modelInfo.name} - ${fieldInfo.name}`);
    }
  }
});
