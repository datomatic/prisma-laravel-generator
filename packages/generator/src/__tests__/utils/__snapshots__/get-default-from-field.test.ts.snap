// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDefaultFromField - mongodb: MongoDbNow - created_at 1`] = `
Object {
  "defaultValue": "Carbon::now()",
  "imports": Set {
    "Illuminate\\\\Support\\\\Carbon",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField - mongodb: MongoDbNow - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Category - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Category - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Category - posts 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - age 1`] = `
Object {
  "defaultValue": 18,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - bool 1`] = `
Object {
  "defaultValue": true,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - email 1`] = `
Object {
  "defaultValue": "hello@world.test",
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - name 1`] = `
Object {
  "defaultValue": "N.A.",
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - now 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: DefaultValues - role 1`] = `
Object {
  "defaultValue": "Role::USER",
  "imports": Set {
    "App\\\\Enums\\\\Prisma\\\\Role",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField: DefaultValues - uuid 1`] = `
Object {
  "defaultValue": "Str::orderedUuid()",
  "imports": Set {
    "Illuminate\\\\Support\\\\Str",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField: FieldComments - age 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: FieldComments - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: FieldComments - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: FieldComments - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: FieldMultipleComments - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: FieldMultipleComments - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Fillable - age 1`] = `
Object {
  "defaultValue": 18,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Fillable - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Fillable - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Fillable - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Fillable - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Guarded - age 1`] = `
Object {
  "defaultValue": 18,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Guarded - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Guarded - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Guarded - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Guarded - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Hidden - api_key 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Hidden - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Hidden - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Hidden - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Hidden - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Ignore - age 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Ignore - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Ignore - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MassAssignable - age 1`] = `
Object {
  "defaultValue": 18,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MassAssignable - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MassAssignable - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MassAssignable - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MassAssignable - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ModelComments - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ModelMultipleComments - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleComments - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleComments - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique - age 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique - firstname 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique - lastname 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique2 - age 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique2 - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique2 - firstname 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique2 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: MultipleUnique2 - lastname 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: NullableField - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: NullableField - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - author 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - authorId 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - categories 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - createdAt 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - published 1`] = `
Object {
  "defaultValue": false,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - title 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - type 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Post - updatedAt 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Profile - bio 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Profile - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Profile - user 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Profile - userId 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ReadOnlyField - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ReadOnlyField - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ReadOnlyField - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ReadOnlyField - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: ReadOnlyField - phone 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete1 - deleted_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete1 - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete1 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete1 - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete2 - deleted_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete2 - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete2 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: SoftDelete2 - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps1 - created_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps1 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps1 - updated_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps2 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps2 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps2 - updated_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps3 - created_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps3 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps3 - updated_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps4 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps4 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps4 - updated_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps5 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps5 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps5 - updated_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps6 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps6 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps6 - updated_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps7 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps7 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps7 - updated_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps8 - created_at 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps8 - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Timestamps8 - updated_timestamp 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - active 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - age 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - birth 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - blob 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - price 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - schema 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - seconds 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Types - speed 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Unique - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Unique - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Unique - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Unique - username 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - posts 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - profile 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: User - role 1`] = `
Object {
  "defaultValue": "Role::USER",
  "imports": Set {
    "App\\\\Enums\\\\Prisma\\\\Role",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField: Visible - api_key 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Visible - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Visible - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Visible - name 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: Visible - password 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithArray - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithArray - numbers 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithBackedEnum - array 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithBackedEnum - defaultEnum 1`] = `
Object {
  "defaultValue": "BackedEnum::A",
  "imports": Set {
    "App\\\\Enums\\\\Prisma\\\\BackedEnum",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField: WithBackedEnum - enum 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithBackedEnum - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithBackedEnum - nullable 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithDbType - email 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithDbType - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithEnum - array 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithEnum - defaultEnum 1`] = `
Object {
  "defaultValue": "Enum::A",
  "imports": Set {
    "App\\\\Enums\\\\Prisma\\\\Enum",
  },
  "isMethodCall": true,
}
`;

exports[`getDefaultFromField: WithEnum - enum 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithEnum - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithEnum - nullable 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithGlobalPrimaryKey - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithGlobalPrimaryKeyDifferentName - primary_id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithGlobalStringPrimaryKey - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithGlobalStringPrimaryKeyDifferentName - primary_id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithPrimaryKey - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithPrimaryKeyDifferentName - primary_id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithStringPrimaryKey - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithStringPrimaryKeyDifferentName - primary_id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;

exports[`getDefaultFromField: WithTableMap - id 1`] = `
Object {
  "defaultValue": undefined,
  "imports": Set {},
  "isMethodCall": false,
}
`;
